plotly/tests/test_core/test_tools/test_TraceFactory.py

```diff
@@ -0,0 +1,112 @@
+                                      2.0677536925644366, 2.0909297426825684,
+                                      2.051107819102551, None],
+                                'x': [1, 1.0540302305868139, None, 2,
+                                      1.9583853163452858, None,
+                                      1.052143029378767, 1.0540302305868139,
+                                      1.0184841899864512, None,
+                                      1.9909870141679737, 1.9583853163452858,
+                                      1.9546151170949464, None],
+                                'line': {'color': 'purple', 'width': 3},
+                                'type': 'scatter', 'mode': 'lines', })
+
+
+@raises(ValueError)
+def test_wrong_arrow_scale():
+    stream = tls.TraceFactory.create_streamline(x=[0, 2], y=[0, 2],
+                                                u=[[-1, -5], [-1, -5]],
+                                                v=[[1, 1], [-3, -3]],
+                                                arrow_scale=0)
+
+
+@raises(ValueError)
+def test_wrong_density():
+    stream = tls.TraceFactory.create_streamline(x=[0, 2], y=[0, 2],
+                                                u=[[-1, -5], [-1, -5]],
+                                                v=[[1, 1], [-3, -3]],
+                                                density=-1)
+
+
+@raises(PlotlyError)
+def test_uneven_x():
+    stream = tls.TraceFactory.create_streamline(x=[0, 2, 7, 9], y=[0, 2, 4, 6],
+                                                u=[[-1, -5], [-1, -5]],
+                                                v=[[1, 1], [-3, -3]])
+
+
+@raises(PlotlyError)
+def test_uneven_y():
+    stream = tls.TraceFactory.create_streamline(x=[0, 2, 4], y=[0, 2, 6],
+                                                u=[[-1, -5], [-1, -5]],
+                                                v=[[1, 1], [-3, -3]])
+
+
+class TestStreamline(TestCase):
+
+    def test_simple(self):
+        self.assertEqual((tls.TraceFactory.create_streamline(x=[0, 2],
+                                                             y=[0, 2],
+                                                             u=[[-1, -5],
+                                                                [-1, -5]],
+                                                             v=[[1, 1],
+                                                                [-3, -3]],
+                                                             density=2,
+                                                             arrow_scale=.4,
+                                                             angle=math.pi/6,
+                                                             line=Line(
+                                                              color='purple',
+                                                              width=3))).keys(),
+                         (['y', 'x', 'line', 'type', 'mode']))
```

Comment:
:tiger2: this doesn't really test that `create_streamline` works. Mind adding a few more tests? These should really include bad param tests to make sure we're not blindly returning garbage out when there's garbage in.